### PR TITLE
Issue1 upgrade java and gradle

### LINKED ISSUE #1: Upgrade java to 17 and gradle to 7.x

### PR DESCRIPTION
**Pull request by shvets-semandex**
closes #1, #2

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -2,13 +2,6 @@
...
-group 'mbuhot'
-version '0.8.0'
-
-apply plugin: 'kotlin'
-apply plugin: 'com.jfrog.bintray'
-apply plugin: 'maven-publish'
-
 buildscript {
     repositories {
         mavenCentral()
```
Comment (chirag4semandex) [inline build.gradle:7]: Lets remove the old `jcenter()` dependency as well

File: build.gradle [hunk 2]
```diff
@@ -20,21 +13,77 @@ buildscript {
...
+apply plugin: 'maven-publish'
 
 repositories {
     mavenCentral()
+
+    maven {
+        def repo_url = project.hasProperty('repo_url') ? project.property('repo_url')
+                : 'https://nexus.semandex.net:44355/repository/semandex-maven-prod/'
+        url repo_url
+        allowInsecureProtocol = true
```
Comment (chirag4semandex) [inline build.gradle:29]: We dont need this anymore since our repo is secured now. Please take it out and check if it works

File: build.gradle [hunk 3]
```diff
@@ -20,21 +13,77 @@ buildscript {
...
+                username = nexus_user
+                password = nexus_pass
+            }
+        }
+    }
+}
+
+
+ext {
+  kotlinVersion = '1.3.61'
```
Comment (chirag4semandex) [inline build.gradle:68]: Isnt this an old version of kotlin? We cant upgrade right now?

File: build.gradle (hunk 2 above)
Comment (shvets-semandex) [reply build.gradle:29]: this is what we have in gradle-script-plugin now. It will be replaced soon. let's keep it as-is

File: build.gradle (hunk 3 above)
Comment (shvets-semandex) [reply build.gradle:68]: no. only with v2 of opensearch

**Review by chirag4semandex — APPROVED** [review]

**Review by djmakwana — APPROVED** [review]

**Pull request merged**